// @ant-design/pro-layout/es/PageContainer/index.js
import "antd/es/affix/style";
import _Affix from "antd/es/affix";
import "antd/es/page-header/style";
import _PageHeader from "antd/es/page-header";
import "antd/es/tabs/style";
import _Tabs from "antd/es/tabs";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React, { useContext } from 'react';
import classNames from 'classnames';
import RouteContext from '../RouteContext';
import GridContent from '../GridContent';
import FooterToolbar from '../FooterToolbar';
import './index.less';
/**
 * render Footer tabList
 * In order to be compatible with the old version of the PageHeader
 * basically all the functions are implemented.
 */

var renderFooter = function renderFooter(_ref) {
  var tabList = _ref.tabList,
      tabActiveKey = _ref.tabActiveKey,
      onTabChange = _ref.onTabChange,
      tabBarExtraContent = _ref.tabBarExtraContent,
      tabProps = _ref.tabProps,
      prefixedClassName = _ref.prefixedClassName;

  if (tabList && tabList.length) {
    return /*#__PURE__*/React.createElement(_Tabs, _extends({
      className: "".concat(prefixedClassName, "-tabs"),
      activeKey: tabActiveKey,
      onChange: function onChange(key) {
        if (onTabChange) {
          onTabChange(key);
        }
      },
      tabBarExtraContent: tabBarExtraContent
    }, tabProps), tabList.map(function (item, index) {
      return (
        /*#__PURE__*/
        // eslint-disable-next-line react/no-array-index-key
        React.createElement(_Tabs.TabPane, _extends({}, item, {
          tab: item.tab,
          key: item.key || index
        }))
      );
    }));
  }

  return null;
};

var renderPageHeader = function renderPageHeader(content, extraContent, prefixedClassName) {
  if (!content && !extraContent) {
    return null;
  }

  return /*#__PURE__*/React.createElement("div", {
    className: "".concat(prefixedClassName, "-detail")
  }, /*#__PURE__*/React.createElement("div", {
    className: "".concat(prefixedClassName, "-main")
  }, /*#__PURE__*/React.createElement("div", {
    className: "".concat(prefixedClassName, "-row")
  }, content && /*#__PURE__*/React.createElement("div", {
    className: "".concat(prefixedClassName, "-content")
  }, content), extraContent && /*#__PURE__*/React.createElement("div", {
    className: "".concat(prefixedClassName, "-extraContent")
  }, extraContent))));
};

var defaultPageHeaderRender = function defaultPageHeaderRender(props, value) {
  var title = props.title,
      content = props.content,
      pageHeaderRender = props.pageHeaderRender,
      header = props.header,
      extraContent = props.extraContent,
      style = props.style,
      prefixCls = props.prefixCls,
      restProps = _objectWithoutProperties(props, ["title", "content", "pageHeaderRender", "header", "extraContent", "style", "prefixCls"]);

  if (pageHeaderRender) {
    return pageHeaderRender(_objectSpread(_objectSpread({}, props), value));
  }

  var pageHeaderTitle = title;

  if (!title && title !== false) {
    pageHeaderTitle = value.title;
  }

  return /*#__PURE__*/React.createElement(_PageHeader, _extends({}, value, {
    title: pageHeaderTitle
  }, restProps, {
    footer: renderFooter(_objectSpread(_objectSpread({}, restProps), {}, {
      prefixedClassName: value.prefixedClassName
    }))
  }, header, {
    prefixCls: prefixCls
  }), renderPageHeader(content, extraContent, value.prefixedClassName));
};

var PageContainer = function PageContainer(props) {
  var children = props.children,
      style = props.style,
      footer = props.footer,
      ghost = props.ghost,
      fixedHeader = props.fixedHeader,
      _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'ant-pro' : _props$prefixCls;
  var value = useContext(RouteContext);
  var prefixedClassName = "".concat(prefixCls, "-page-container");
  var className = classNames(prefixedClassName, props.className, _defineProperty({}, "".concat(prefixCls, "-page-container-ghost"), ghost));
  var headerDom = /*#__PURE__*/React.createElement("div", {
    className: "".concat(prefixedClassName, "-warp")
  }, defaultPageHeaderRender(props, _objectSpread(_objectSpread({}, value), {}, {
    prefixCls: undefined,
    prefixedClassName: prefixedClassName
  })));
  return /*#__PURE__*/React.createElement("div", {
    style: style,
    className: className
  }, fixedHeader ? /*#__PURE__*/React.createElement(_Affix, {
    offsetTop: value.hasHeader ? value.headerHeight : 0
  }, headerDom) : headerDom, /*#__PURE__*/React.createElement(GridContent, null, children ? /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("div", {
    className: "".concat(prefixedClassName, "-children-content")
  }, children), value.hasFooterToolbar && /*#__PURE__*/React.createElement("div", {
    style: {
      height: 48,
      marginTop: 24
    }
  })) : null), footer && /*#__PURE__*/React.createElement(FooterToolbar, null, footer));
};

export default PageContainer;
